### PR TITLE
Photo search limit

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Needed for https://github.com/nextcloud/android/pull/2250

TODO
- [x] split it up so that search limit with timestamp works
- [x] base other search (as new PR) on this PR

This is still wip as the server has some problems.
While the linked PR is nice to have, this is not needed for this PR, as limit will work.
For #2250 we will just query 50 images, and for next query also 50 images but with timestamp limitation.

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
:+1:

**Comment by AndyScherzinger** [comment]
Needs a rebase now that other PRs have been merged

**Comment by tobiasKaminsky** [comment]
It is still under development ;-)
As this requires also some webdav changes, this will go into NC14.

**Comment by tobiasKaminsky** [comment]
@rullzer webdav search on server still does not have reliable pagination / limit, right?
How can we enhance this otherwise?

**Comment by mario** [comment]
It will have it in 15 @tobiasKaminsky as @icewind1991 did work on this.

**Comment by tobiasKaminsky** [comment]
@mario @icewind1991 do you have a PR link for this? So i can test it :+1:

**Comment by AndyScherzinger** [comment]
@rullzer @mario @icewind1991 anyone having a PR link for the server change to @tobiasKaminsky can test it? :)

**Comment by tobiasKaminsky** [comment]
@mario @icewind1991 which PR do you mean?

**Comment by tobiasKaminsky** [comment]
Might be this one? https://github.com/nextcloud/server/pull/10880

**Comment by mario** [comment]
Yes

File: src/androidTest/java/com/owncloud/android/AbstractIT.java [hunk 1]
```diff
@@ -43,6 +51,43 @@ public static void beforeAll() {
...
+     * @param context  Android context to access assets and file system.
+     * @return File instance of the extracted file.
+     */
+    public static File extractAsset(String fileName, Context context) throws IOException {
+        File extractedFile = new File(context.getCacheDir() + File.separator + fileName);
+        if (!extractedFile.exists()) {
+            InputStream in = null;
+            FileOutputStream out = null;
+            in = context.getAssets().open(fileName);
+            out = new FileOutputStream(extractedFile);
```
Comment (AndyScherzinger) [inline src/androidTest/java/com/owncloud/android/AbstractIT.java]: why do you assign null right before assigning a new reference to in/out variables?

Comment (tobiasKaminsky) [reply src/androidTest/java/com/owncloud/android/AbstractIT.java]: Nice founding.
I have no idea what I did there :roll_eyes:

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky test cases keep failing (probably need an update?)

File: src/androidTest/java/com/owncloud/android/AbstractIT.java [hunk 2]
```diff
@@ -43,6 +51,54 @@ public static void beforeAll() {
...
 
+    public String createFile(String name) throws IOException {
+        File tempDir = context.getExternalCacheDir();
+
+        if (tempDir == null) {
+            throw new IOException("Temp dir is null");
+        }
+
+        if (!tempDir.exists()) {
+            if (!tempDir.mkdirs()) {
```
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/androidTest/java/com/owncloud/android/AbstractIT.java]: valid point 👍

**Comment by AndyScherzinger** [comment]
Tests still failing and one (valid) codacy warning

**Pull request merged**